spk/duplicity/src/service-setup.sh

```diff
@@ -0,0 +1,26 @@
+
+### service-setup.sh
+
+PYTHON_DIR="/usr/local/python3"
+VIRTUALENV="${PYTHON_DIR}/bin/virtualenv"
+PATH="${SYNOPKG_PKGDEST}/env/bin:${SYNOPKG_PKGDEST}/bin:${PYTHON_DIR}/bin:${PATH}"
+
+service_postinst ()
+{
+    # Create a Python virtualenv
+    ${VIRTUALENV} --system-site-packages ${SYNOPKG_PKGDEST}/env >> ${INST_LOG}
+
+    # Install the wheels
+    ${SYNOPKG_PKGDEST}/env/bin/pip install --no-deps --no-input --upgrade --no-index --find-links ${SYNOPKG_PKGDEST}/share/wheelhouse ${SYNOPKG_PKGDEST}/share/wheelhouse/*.whl >> ${INST_LOG} 2>&1
+}
+
+
+# use alternate TMPDIR for service_postinst
+# /tmp might be too small and you get errors like:
+# Could not install packages due to an EnvironmentError: [Errno 28] No space left on device
+TMPDIR=${SYNOPKG_PKGDEST}/tmp
+if [[ ! -e "${TMPDIR}" ]]; then
+    mkdir -p "${TMPDIR}"
+    chown ${EFF_USER} "${TMPDIR}"
+fi
+export TMPDIR=${SYNOPKG_PKGDEST}/tmp
```

Comment:
I am just wondering if it would not make sense to move this TMPDIR logic into generic framework installer as a function to call, to avoid such a copy-paste pattern